testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/AnotherSseResource.java

```diff
@@ -25,24 +25,33 @@
    @Path("/subscribe")
    @Produces(MediaType.SERVER_SENT_EVENTS)
    public void subscribe(@Context SseEventSink sink) throws IOException {
-	   if (sink == null) {
-           throw new IllegalStateException("No client connected.");
-       }
-       //subscribe
-	   if (sseBroadcaster == null) {
-		   sseBroadcaster = sse.newBroadcaster();
-	   }
-	   sseBroadcaster.register(sink);	  
+      synchronized (outputLock)
+      {
+         if (sink == null)
+         {
+            throw new IllegalStateException("No client connected.");
+         }
+      }
+      eventSink = sink;
+      //subscribe
+      if (sseBroadcaster == null)
+      {
+         sseBroadcaster = sse.newBroadcaster();
+      }
+      sseBroadcaster.register(sink);  
    }
   
    @DELETE
-   public void close() throws IOException {
+   @Produces(MediaType.TEXT_PLAIN)
+   public boolean close() throws IOException {
        synchronized (outputLock) {
            if (eventSink != null) {
-               eventSink.close();
-               eventSink = null;
+              try (SseEventSink sink = eventSink) {
```

Comment:
While this is good and it's one of the reasons for this PR, I think you should add a comment saying something like "we could have simply invoked eventSink.close(), verifying auto-closeable functionalities though". This would help users understand the test more easily (the test here is becoming complex).